curp/src/rpc/mod.rs

```diff
@@ -208,65 +198,75 @@ impl WaitSyncedResponse {
-            ), // this is highly unlikely to happen,
-            (Some(Err(err)), None) => {
-                WaitSyncedResponse::new_error(CommandSyncError::<C>::Execute(err))
-            }
-            // The er is ignored as the propose has failed
-            (Some(Ok(_er)), Some(Err(err))) => {
-                WaitSyncedResponse::new_error(CommandSyncError::<C>::AfterSync(err))
-            }
-            (Some(Ok(er)), Some(Ok(asr))) => WaitSyncedResponse::new_success::<C>(&asr, &er),
-            // The er is ignored as the propose has failed
-            (Some(Ok(_er)), None) => {
-                WaitSyncedResponse::new_error::<C>(
-                    WaitSyncError::Other("can't get after sync result".to_owned()).into(),
-                ) // this is highly unlikely to happen,
+            (Ok(ref er), Some(Err(ref asr_err))) => {
+                WaitSyncedResponse::new_asr_error::<C>(er, asr_err)
             }
+            (Ok(ref er), Some(Ok(ref asr))) => WaitSyncedResponse::new_success::<C>(asr, er),
+            (Ok(ref _er), None) => unreachable!("can't get after sync result"),
+            (Err(ref err), _) => WaitSyncedResponse::new_er_error::<C>(err),
         }
     }
 
     /// Into deserialized result
     pub(crate) fn into<C: Command>(self) -> Result<SyncResult<C>, PbSerializeError> {
-        let res = match self.sync_result {
-            None => unreachable!("WaitSyncedResponse should contain valid sync_result"),
-            Some(SyncResultRaw::Success(success)) => SyncResult::Success {
-                asr: <C as Command>::ASR::decode(&success.after_sync_result)?,
-                er: <C as Command>::ER::decode(&success.exe_result)?,
-            },
-            Some(SyncResultRaw::Error(err)) => {
-                let cmd_sync_err = err
-                    .command_sync_error
-                    .ok_or(PbSerializeError::EmptyField)?
-                    .try_into()?;
-                SyncResult::Error(cmd_sync_err)
+        let res = match (self.exe_result, self.after_sync_result) {
+            (None, _) => unreachable!("WaitSyncedResponse should contain a valid exe_result"),
+            (Some(er), None) => {
+                if let Some(CmdResultInner::Error(buf)) = er.result {
+                    SyncResult::Error(<C as Command>::Error::decode(buf.as_slice())?)
+                } else {
+                    unreachable!("er should not be None")
+                }
+            }
+            (Some(er), Some(asr)) => {
+                let er = if let Some(CmdResultInner::Ok(er)) = er.result {
+                    <C as Command>::ER::decode(er.as_slice())?
+                } else {
+                    unreachable!("")
+                };
+                let asr = if let Some(CmdResultInner::Ok(asr)) = asr.result {
+                    <C as Command>::ASR::decode(asr.as_slice())?
+                } else {
+                    unreachable!("")
```

Comment:
These two unreachable does not have panic message.

Reply:
Done